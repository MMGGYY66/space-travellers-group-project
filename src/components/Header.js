import { Link, NavLink, useLocation } from 'react-router-dom';

const Header = () => {
  const links = [
    {
      id: 1,
      title: 'Rockets',
      to: '/Rockets',
    },
    {
      id: 2,
      title: 'Missions',
      to: '/Missions',
    },
    {
      id: 3,
      title: 'My Profile',
      to: '/Profile',
    },
  ];

  const location = useLocation();
  return (
    <header className="header-container">
      <nav className="navbar bg-white py-4 px-5 nav-style">
        <img src="/planet.png" alt="Logo" width={50} />
        <h3 className="mt-2">
          <Link to="/" className="nav-brand">Space Travelers&apos; Hub</Link>
        </h3>

        <nav className="nav-container">
          <ul className="links-list">
            {links.length > 0
              && links.map((link) => (
                <li className={`link-item ${link.to === location.pathname ? 'link-item-active' : ''}`} key={link.id}>
                  <NavLink to={link.to}>
                    {link.title}
                  </NavLink>
                </li>
              ))}
          </ul>

        </nav>
      </nav>
    </header>
  );
};

export default Header;
