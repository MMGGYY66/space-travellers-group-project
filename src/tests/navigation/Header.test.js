import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../../components/Header';

describe('Header Component', () => {
  afterEach(cleanup);

  test('Renders the logo', () => {
    render(
      <Router>
        <Header />
      </Router>,
    );
    const logoImage = screen.getByAltText(/logo/i);
    expect(logoImage).toBeInTheDocument();
  });

  test('Renders the Header component', () => {
    const { getByText } = render(
      <Router>
        <Header />
      </Router>,
    );
    const linkElement = getByText(/Missions/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('Builds the snapshot of the Header component', () => {
    const navigation = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(navigation).toMatchSnapshot();
  });
});
