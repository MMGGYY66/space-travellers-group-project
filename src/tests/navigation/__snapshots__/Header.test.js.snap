// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Builds the snapshot of the Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="d-flex justify-content-between border-bottom nav-head px-5 mt-3"
      >
        <div
          class="d-flex align-middle gap-3"
        >
          <img
            alt="logo"
            src="planet.png"
          />
          <h3
            class="mt-2"
          >
            Space Travelers' Hub
          </h3>
        </div>
        <ul
          class="d-flex justify-content-around align-items-center list-unstyled gap-3 mt-3"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="inactive"
              href="/missions"
            >
              Missions
            </a>
          </li>
          |
          <li>
            <a
              class="inactive"
              href="/my_profile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="d-flex justify-content-between border-bottom nav-head px-5 mt-3"
    >
      <div
        class="d-flex align-middle gap-3"
      >
        <img
          alt="logo"
          src="planet.png"
        />
        <h3
          class="mt-2"
        >
          Space Travelers' Hub
        </h3>
      </div>
      <ul
        class="d-flex justify-content-around align-items-center list-unstyled gap-3 mt-3"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="inactive"
            href="/missions"
          >
            Missions
          </a>
        </li>
        |
        <li>
          <a
            class="inactive"
            href="/my_profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
